src/main/kotlin/se/uulm/snowballr/backend/plugins/Plugin.kt

```diff
@@ -0,0 +1,168 @@
+     * while this function waits for the correct [Return] type, they will be ignored by this function.
+     *
+     * Prefer the generic inline function without a klass parameter instead of this variant.
+     *
+     * @param Return The type of the event expected as a response.
+     * @param klass The [KClass] of the expected return type [Return].
+     * @param request The event to be sent to the event transport.
+     * @throws PluginClosedException If the plugin is closed.
+     * @return The response event of type [Return].
+     */
+    @Suppress("UNCHECKED_CAST")
+    suspend fun <Return : Event> call(klass: KClass<Return>, request: Event): Return {
+        throwIfClosed()
+        var queue = callHandlers.get(klass)
+        if (queue == null) {
+            queue = LinkedList()
+            callHandlers.put(klass, queue)
+        }
+
+        val returnValue = suspendCoroutine<Return> { continuation ->
+            queue.add { event ->
+                continuation.resumeWith(Result.success(event as Return))
+            }
+            runBlocking { eventTransport.send(request) }
+        }
+
+        return returnValue
+    }
+
+    /**
+     * Registers a handler for a specific event type. That handler is called when an event of the specific class is
+     * received. There can only be one handler for one specific event type at a time.
+     *
+     * The handler will be called from another thread.
+     *
+     * The handler will be called from another thread.
+     *
+     * @param E The type of event to register this handler for. Must be a subclass of Event.
+     * @param handler A lambda function that processes an instance of the specified class type.
+     * @throws PluginClosedException If the plugin is closed.
+     */
+    @Suppress("UNCHECKED_CAST")
+    inline fun <reified E : Event> handle(noinline handler: suspend (event: E) -> Unit) {
+        handle(
+            E::class,
+            handler as suspend (Any) -> Unit,
+        )
+    }
+
+    /**
+     * Registers a handler for a specific event type. That handler is called when an event of the specific class is
+     * received. There can only be one handler for one specific event type at a time.
+     *
+     * The handler will be called from another thread.
+     *
+     * Prefer the generic inline function without a klass parameter instead of this variant.
+     *
+     * @param klass The class type for which the handler is being registered.
+     * @param handler A lambda function that processes an instance of the specified class type.
+     * @throws PluginClosedException If the plugin is closed.
+     */
+    fun handle(klass: KClass<*>, handler: suspend (Any) -> Unit) {
+        throwIfClosed()
+        handlers.put(klass, handler)
+    }
+
+    private fun throwIfClosed() {
+        if (isClosed) throw PluginClosedException()
+    }
+
+    class PluginClosedException : Exception("The Plugin is closed and no methods may be called on it")
```

Comment:
How can a plugin be closed?
```suggestion
    class PluginClosedException : Exception("The plugin is not running and no methods can be called on it.")
```

Reply:
A fetcher should be closed when removed. This functionality was missing from the `FetcherManager` and should now be implemented.